### PR TITLE
add new getter GetMapSlice

### PR DESCRIPTION
**Pull request by s-kostyaev**

### CODE REVIEW

File: slice.go [hunk 1]
```diff
@@ -123,6 +123,51 @@ func (h Hash) GetStringSlice(path ...string) ([]string, error) {
...
+	if m == nil {
+		return []map[string]interface{}{}, notFoundError{path}
+	}
+	switch val := m.(type) {
+	case []map[string]interface{}:
+		return val, nil
+	case []interface{}:
+		result := []map[string]interface{}{}
+		for _, elem := range val {
+			switch e := elem.(type) {
```
Comment (kovetskiy) [inline slice.go]: Can you rename one-letter variable names?

**Comment by s-kostyaev** [comment]
@kovetskiy, fixed

File: slice.go [hunk 2]
```diff
@@ -123,6 +123,51 @@ func (h Hash) GetStringSlice(path ...string) ([]string, error) {
...
+				if newKey, ok := key.(string); ok {
+					newMap[newKey] = value
+				}
+			}
+			result = append(result, newMap)
+		}
+		return result, nil
+	default:
+		return []map[string]interface{}{}, fmt.Errorf(
+			"cannot convert %s []map[string]interface{}", strings.Join(path, "."),
```
Comment (kovetskiy) [inline slice.go]: cannot convert %s to ...?

File: slice.go [hunk 3]
```diff
@@ -177,3 +222,17 @@ func (h Hash) AppendStringSlice(val string, path ...string) error {
 	h.Set(slice, path...)
 	return nil
 }
+
+func (hash Hash) AppendMapSlice(val map[string]interface{}, path ...string) error {
+	slice, err := hash.GetMapSlice(path...)
+	if err != nil {
```
Comment (kovetskiy) [inline slice.go]: ```if err != nil && !IsNotFound(err) {```

File: slice.go [hunk 4]
```diff
@@ -123,6 +123,51 @@ func (h Hash) GetStringSlice(path ...string) ([]string, error) {
...
+				result = append(result, typedElem)
+			case map[interface{}]interface{}:
+				newMap := make(map[string]interface{})
+				for key, value := range typedElem {
+					if newKey, ok := key.(string); ok {
+						newMap[newKey] = value
+					}
+				}
+				result = append(result, newMap)
+			default:
```
Comment (kovetskiy) [inline slice.go]: where is `default:` body?

**Comment by s-kostyaev** [comment]
@kovetskiy, fixed

File: slice.go [hunk 5]
```diff
@@ -123,6 +123,52 @@ func (h Hash) GetStringSlice(path ...string) ([]string, error) {
...
+			case map[string]interface{}:
+				result = append(result, typedElem)
+			case map[interface{}]interface{}:
+				newMap := make(map[string]interface{})
+				for key, value := range typedElem {
+					if newKey, ok := key.(string); ok {
+						newMap[newKey] = value
+					}
+				}
+				result = append(result, newMap)
```
Comment (seletskiy) [inline slice.go]: Why not just recurse there?

File: slice_test.go [hunk 6]
```diff
@@ -81,6 +82,49 @@ func TestGetStringSlice(t *testing.T) {
...
+	err := yaml.Unmarshal([]byte(testYaml), &rawMap)
+	if err != nil {
+		t.Errorf("Error yaml unmarshall: %s", err.Error())
+	}
+
+	hash := HashFromMap(rawMap)
+
+	result, err := hash.GetMapSlice("deploy", "chmod")
+	if err != nil {
+		t.Errorf("Error getting map slice: %s", err.Error())
```
Comment (kovetskiy) [inline slice_test.go]: should be fatalf, I guess

Comment (s-kostyaev) [inline slice_test.go]: fixed

**Comment by kovetskiy** [comment]
@s-kostyaev please do not use ambigious commit messages like `fix test`. in this case it should be `tests: using t.Fatalf instead of t.Errorf`

**Comment by s-kostyaev** [comment]
@kovetskiy fixed

**Pull request merged**